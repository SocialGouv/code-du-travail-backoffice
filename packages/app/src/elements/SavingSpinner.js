// https://loading.io/css/

import { keyframes } from "@emotion/core";
import styled from "@emotion/styled";
import PropTypes from "prop-types";
import React from "react";

const Container = styled.div`
  display: inline-block;
  height: ${p => p.size}px;
  position: relative;
  width: ${p => p.size}px;
`;

const rotation = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
`;
const Spinner = styled.div`
  animation: ${rotation} 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
  border: ${p => Math.round(p.size / 10)}px solid ${p => p.color};
  border-color: ${p => p.color} transparent transparent transparent;
  border-radius: 50%;
  box-sizing: border-box;
  display: block;
  height: ${p => Math.round((p.size * 4) / 5)}px;
  margin: ${p => Math.round(p.size / 10)}px;
  position: absolute;
  width: ${p => Math.round((p.size * 4) / 5)}px;
`;
const FirstSpinner = styled(Spinner)`
  animation-delay: -0.45s;
`;
const SecondSpinner = styled(Spinner)`
  animation-delay: -0.3s;
`;
const ThirdSpinner = styled(Spinner)`
  animation-delay: -0.15s;
`;

const SavingSpinner = ({ color = "white", size = 64, ...props }) => {
  return (
    <Container size={size} {...props}>
      <FirstSpinner color={color} data-testid="first-spinner" size={size} />
      <SecondSpinner color={color} data-testid="second-spinner" size={size} />
      <ThirdSpinner color={color} data-testid="third-spinner" size={size} />
      <Spinner color={color} data-testid="spinner" size={size} />
    </Container>
  );
};

SavingSpinner.propTypes = {
  color: PropTypes.string,
  size: PropTypes.number,
};

export default SavingSpinner;
