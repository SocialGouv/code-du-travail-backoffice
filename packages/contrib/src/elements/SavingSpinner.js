// https://loading.io/css/

import { keyframes } from "@emotion/core";
import styled from "@emotion/styled";
import React from "react";

const Container = styled.div`
  display: inline-block;
  height: ${props => props.size}px;
  position: relative;
  width: ${props => props.size}px;
`;

const rotation = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
`;
const Spinner = styled.div`
  animation: ${rotation} 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
  border: ${props => Math.round(props.size / 10)}px solid ${props => props.color};
  border-color: ${props => props.color} transparent transparent transparent;
  border-radius: 50%;
  box-sizing: border-box;
  display: block;
  height: ${props => Math.round((props.size * 4) / 5)}px;
  margin: ${props => Math.round(props.size / 10)}px;
  position: absolute;
  width: ${props => Math.round((props.size * 4) / 5)}px;
`;
const FirstSpinner = styled(Spinner)`
  animation-delay: -0.45s;
`;
const SecondSpinner = styled(Spinner)`
  animation-delay: -0.3s;
`;
const ThirdSpinner = styled(Spinner)`
  animation-delay: -0.15s;
`;

export default ({ color, size }) => {
  const _color = color !== undefined ? color : "white";
  const _size = size !== undefined ? size : 64;

  return (
    <Container size={_size}>
      <FirstSpinner color={_color} size={_size} />
      <SecondSpinner color={_color} size={_size} />
      <ThirdSpinner color={_color} size={_size} />
      <Spinner color={_color} size={_size} />
    </Container>
  );
};
