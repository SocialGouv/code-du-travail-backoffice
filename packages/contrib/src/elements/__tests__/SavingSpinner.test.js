import React from "react";
import { render } from "react-testing-library";
import SavingSpinner from "../SavingSpinner";

describe("[Contrib] elements/<SavingSpinner />", () => {
  it("should match snapshot", () => {
    const { container } = render(<SavingSpinner />);

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot", () => {
    const { container } = render(<SavingSpinner color="red" />);

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot", () => {
    const { container } = render(<SavingSpinner size="32" />);

    expect(container).toMatchSnapshot();
  });
});
