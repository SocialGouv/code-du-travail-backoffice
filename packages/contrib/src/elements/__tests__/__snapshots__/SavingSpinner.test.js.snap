// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Contrib] elements/<SavingSpinner /> should pass 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  100% {
    -webkit-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes animation-0 {
  0% {
    -webkit-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  100% {
    -webkit-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes animation-0 {
  0% {
    -webkit-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  100% {
    -webkit-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes animation-0 {
  0% {
    -webkit-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  100% {
    -webkit-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

.emotion-8 {
  display: inline-block;
  height: 64px;
  position: relative;
  width: 64px;
}

.emotion-0 {
  -webkit-animation: animation-0 1.2s cubic-bezier(0.5,0,0.5,1) infinite;
  animation: animation-0 1.2s cubic-bezier(0.5,0,0.5,1) infinite;
  border: 6px solid white;
  border-color: white transparent transparent transparent;
  border-radius: 50%;
  box-sizing: border-box;
  display: block;
  height: 51px;
  margin: 6px;
  position: absolute;
  width: 51px;
  -webkit-animation-delay: -0.45s;
  animation-delay: -0.45s;
}

.emotion-2 {
  -webkit-animation: animation-0 1.2s cubic-bezier(0.5,0,0.5,1) infinite;
  animation: animation-0 1.2s cubic-bezier(0.5,0,0.5,1) infinite;
  border: 6px solid white;
  border-color: white transparent transparent transparent;
  border-radius: 50%;
  box-sizing: border-box;
  display: block;
  height: 51px;
  margin: 6px;
  position: absolute;
  width: 51px;
  -webkit-animation-delay: -0.3s;
  animation-delay: -0.3s;
}

.emotion-4 {
  -webkit-animation: animation-0 1.2s cubic-bezier(0.5,0,0.5,1) infinite;
  animation: animation-0 1.2s cubic-bezier(0.5,0,0.5,1) infinite;
  border: 6px solid white;
  border-color: white transparent transparent transparent;
  border-radius: 50%;
  box-sizing: border-box;
  display: block;
  height: 51px;
  margin: 6px;
  position: absolute;
  width: 51px;
  -webkit-animation-delay: -0.15s;
  animation-delay: -0.15s;
}

.emotion-6 {
  -webkit-animation: animation-0 1.2s cubic-bezier(0.5,0,0.5,1) infinite;
  animation: animation-0 1.2s cubic-bezier(0.5,0,0.5,1) infinite;
  border: 6px solid white;
  border-color: white transparent transparent transparent;
  border-radius: 50%;
  box-sizing: border-box;
  display: block;
  height: 51px;
  margin: 6px;
  position: absolute;
  width: 51px;
}

<div
  className="emotion-8 emotion-9"
  size={64}
>
  <div
    className="emotion-0 emotion-1"
    color="white"
    size={64}
  />
  <div
    className="emotion-2 emotion-3"
    color="white"
    size={64}
  />
  <div
    className="emotion-4 emotion-5"
    color="white"
    size={64}
  />
  <div
    className="emotion-6 emotion-7"
    color="white"
    size={64}
  />
</div>
`;
